### ISSUE TITLE
Tenant has been throttled. Too Many Requests.

### ISSUE THREAD

**Issue by hecnavsanz** [oracle/oci-ansible-collection#191]
## Issue Report

"Tenant has been throttled. Too Many Requests." error message when using oracle.oci.oci_identity_tag to delete multiple tags.

**Describe the issue**

Trying to delete multiple tags in a list (with_items) sequentially (wait: yes) gives this error above. Running ansible with -vvv seems to not wait so maybe this is why the request limit is reached due to too many tag deletes. Deleting a tag could take up to 10 mins so (wait_timeout:9000) is used.

**Expected behavior**

For each tag in a list should wait until is DELETED with "wait: yes" but it doesn't seem to.

**Environment**

* OS version: Ubuntu 20.04.5 LTS (focal fossa)

* Ansible version:

ansible [core 2.13.7]
    config file = /etc/ansible/ansible.cfg
    configured module search path = ['/home/oracle/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
    ansible python module location = /home/oracle/oci-ansible-venv/lib/python3.8/site-packages/ansible
    ansible collection location = /home/oracle/.ansible/collections:/usr/share/ansible/collections
    executable location = /home/oracle/oci-ansible-venv/bin/ansible
    python version = 3.8.10 (default, Nov 14 2022, 12:59:47) [GCC 9.4.0]
    jinja version = 3.1.2
    libyaml = True

* OCI Python SDK version:

2.90.0

* OCI Ansible Modules version:

Collection                    Version
----------------------------- -------
amazon.aws                    3.5.0
ansible.netcommon             3.1.3
ansible.posix                 1.4.0
ansible.utils                 2.8.0
ansible.windows               1.12.0
arista.eos                    5.0.1
awx.awx                       21.10.0
azure.azcollection            1.14.0
check_point.mgmt              2.3.0
chocolatey.chocolatey         1.3.1
cisco.aci                     2.3.0
cisco.asa                     3.1.0
cisco.dnac                    6.6.1
cisco.intersight              1.0.22
cisco.ios                     3.3.2
cisco.iosxr                   3.3.1
cisco.ise                     2.5.9
cisco.meraki                  2.13.0
cisco.mso                     2.1.0
cisco.nso                     1.0.3
cisco.nxos                    3.2.0
cisco.ucs                     1.8.0
cloud.common                  2.1.2
cloudscale_ch.cloud           2.2.3
community.aws                 3.6.0
community.azure               1.1.0
community.ciscosmb            1.0.5
community.crypto              2.9.0
community.digitalocean        1.22.0
community.dns                 2.4.2
community.docker              2.7.3
community.fortios             1.0.0
community.general             5.8.3
community.google              1.0.0
community.grafana             1.5.3
community.hashi_vault         3.4.0
community.hrobot              1.6.0
community.libvirt             1.2.0
community.mongodb             1.4.2
community.mysql               3.5.1
community.network             4.0.2
community.okd                 2.2.0
community.postgresql          2.3.1
community.proxysql            1.4.0
community.rabbitmq            1.2.3
community.routeros            2.5.0
community.sap                 1.0.0
community.sap_libs            1.4.0
community.skydive             1.0.0
community.sops                1.5.0
community.vmware              2.10.2
community.windows             1.11.1
community.zabbix              1.9.0
containers.podman             1.10.1
cyberark.conjur               1.2.0
cyberark.pas                  1.0.14
dellemc.enterprise_sonic      1.1.2
dellemc.openmanage            5.5.0
dellemc.os10                  1.1.1
dellemc.os6                   1.0.7
dellemc.os9                   1.0.4
f5networks.f5_modules         1.21.0
fortinet.fortimanager         2.1.7
fortinet.fortios              2.2.1
frr.frr                       2.0.0
gluster.gluster               1.0.2
google.cloud                  1.0.2
hetzner.hcloud                1.9.0
hpe.nimble                    1.1.4
ibm.qradar                    2.1.0
ibm.spectrum_virtualize       1.10.0
infinidat.infinibox           1.3.12
infoblox.nios_modules         1.4.1
inspur.ispim                  1.2.0
inspur.sm                     2.3.0
junipernetworks.junos         3.1.0
kubernetes.core               2.3.2
lowlydba.sqlserver            1.2.0
mellanox.onyx                 1.0.0
netapp.aws                    21.7.0
netapp.azure                  21.10.0
netapp.cloudmanager           21.21.0
netapp.elementsw              21.7.0
netapp.ontap                  21.24.1
netapp.storagegrid            21.11.1
netapp.um_info                21.8.0
netapp_eseries.santricity     1.3.1
netbox.netbox                 3.9.0
ngine_io.cloudstack           2.3.0
ngine_io.exoscale             1.0.0
ngine_io.vultr                1.1.2
openstack.cloud               1.10.0
openvswitch.openvswitch       2.1.0
ovirt.ovirt                   2.4.1
purestorage.flasharray        1.15.0
purestorage.flashblade        1.10.0
purestorage.fusion            1.2.0
sensu.sensu_go                1.13.1
servicenow.servicenow         1.0.6
splunk.es                     2.1.0
t_systems_mms.icinga_director 1.31.4
theforeman.foreman            3.7.0
vmware.vmware_rest            2.2.0
vultr.cloud                   1.3.1
vyos.vyos                     3.0.1
wti.remote                    1.0.4

**Ansible playbook to reproduce the issue**

- name: get compartment facts
  oracle.oci.oci_identity_compartment_facts: 
    name: "{{ ansible_compartment_name }}"
    parent_compartment_id: "{{ tenancy_id }}" # the root compartment
    access_level: ANY
    tenancy: "{{ tenancy_id }}"
    region: "{{ region_name }}"
    lifecycle_state: ACTIVE
    compartment_id_in_subtree: yes
  register: compartment_facts
  tags:
    - get_compartment_facts
- name: get tag namespace facts
  oracle.oci.oci_identity_tag_namespace_facts: 
   compartment_id: "{{ compartment_facts.compartments[0].id }}"
   name: "{{ ansible_tag_namespace_name }}"
   tenancy: "{{ tenancy_id }}"
   region: "{{ region_name }}"
   lifecycle_state: ACTIVE
  register: tag_namespace_facts
  tags:
    - get_tag_namespace_facts
- name: delete tags
      oracle.oci.oci_identity_tag:
        tag_namespace_id: "{{ tag_namespace_facts.tag_namespaces[0].id }}"
        name: "{{ item }}"
        tenancy: "{{ tenancy_id }}"
        region: "{{ region_name }}"
        state: absent
        wait: yes
        wait_timeout: 9000
      with_items: "{{ ansible_tag_names }}"
      tags:
        - delete_tags

where ...

ansible_tag_names: 
      - lab
      - rsrc

**Comment by nikhiban** [comment]
@hecnavsanz Thanks for reporting this. We are checking and will get back to you.

**Comment by dineshsuthar786** [comment]
@hecnavsanz We have added a fix for this in latest release v4.6.0. Kindly check and let us know if you still face the issue.

**Comment by hecnavsanz** [comment]
@dineshsuthar786 I can confirm it's working fine now .. thanks for the update.

OUTPUT

```
TASK [lab-01 : delete tags] ****************************************************************************
changed: [oci-cert] => (item=lab) => {"ansible_loop_var": "item", "changed": true, "item": "lab", "tag": {"compartment_id": "xxxxxxxxxxx", "defined_tags": {"Oracle-Tags": {"CreatedBy": "default/hector.n.sanz@outlook.com", "CreatedOn": "2022-12-24T13:25:37.052Z"}}, "description": "lab", "freeform_tags": {}, "id": "xxxxxxxxxxx", "is_cost_tracking": false, "is_retired": true, "lifecycle_state": "DELETED", "name": "lab", "tag_namespace_id": "xxxxxxxxxxx", "tag_namespace_name": "ansible-tn", "time_created": "2022-12-24T13:25:37.108000+00:00", "validator": null}}
changed: [oci-cert] => (item=rsrc) => {"ansible_loop_var": "item", "changed": true, "item": "rsrc", "tag": {"compartment_id": "xxxxxxxxxx", "defined_tags": {"Oracle-Tags": {"CreatedBy": "default/hector.n.sanz@outlook.com", "CreatedOn": "2022-12-24T13:25:38.737Z"}}, "description": "rsrc", "freeform_tags": {}, "id": "xxxxxxxxxx", "is_cost_tracking": false, "is_retired": true, "lifecycle_state": "DELETED", "name": "rsrc", "tag_namespace_id": "xxxxxxxxxx", "tag_namespace_name": "ansible-tn", "time_created": "2022-12-24T13:25:38.793000+00:00", "validator": null}}

PLAY RECAP *********************************************************************************************
oci-cert                   : ok=9    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
